### PR TITLE
Always use python3 to invoke the integrations downloader

### PR DESCRIPTION
**Pull request by alopezz**
### What does this PR do?

Forces python3 to be used to invoke the downloader regardless of user settings.

### Motivation

We can't support Python2 for the newer versions of the libraries that the downloader depends on. We want to avoid having to maintain two versions (to keep py2 compatibility), and since the downloader script is invoked independently, there's no reason not to use python3 here.
[AITOOLS-77](https://datadoghq.atlassian.net/browse/AITOOLS-77)

### Describe how to test/QA your changes

- Check that the `agent integration install` command keeps working as usual.
- Do the same as in https://github.com/DataDog/datadog-agent/pull/14331#issuecomment-1323315264.
- When a version of the downloader without py2 support is released, check that it works even when the agent is set to use python2 for integrations.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Comment by alopezz** [comment]
I'm confirming that this works after getting an omnibus agent 6 build (a docker build).

I installed explicitly with `--python 2` flag and the output shows that we're using the downloader under `python3.8`:

```
root@1cef53e6ab3b:/# /opt/datadog-agent/bin/agent/agent integration --python 2 -r install datadog-vertica==3.5.0-beta.1
/opt/datadog-agent/embedded/lib/python3.8/site-packages/datadog_checks/downloader/data/repo/targets/simple/datadog-vertica/datadog_vertica-3.5.0b1-py2.py3-none-any.whl
Processing /opt/datadog-agent/embedded/lib/python3.8/site-packages/datadog_checks/downloader/data/repo/targets/simple/datadog-vertica/datadog_vertica-3.5.0b1-py2.py3-none-any.whl
Installing collected packages: datadog-vertica
Successfully installed datadog-vertica-3.5.0b1
Successfully copied configuration file conf.yaml.example
Successfully installed datadog-vertica 3.5.0-beta.1
```

Then, I was able to confirm that the updated version is the one on python2:

```
root@1cef53e6ab3b:/# cat /opt/datadog-agent/embedded/lib/python2.7/site-packages/datadog_checks/vertica/__about__.py
# (C) Datadog, Inc. 2019-present
# All rights reserved
# Licensed under a 3-clause BSD style license (see LICENSE)
__version__ = '3.5.0-beta.1'

root@1cef53e6ab3b:/# cat /opt/datadog-agent/embedded/lib/python3.8/site-packages/datadog_checks/vertica/__about__.py
# (C) Datadog, Inc. 2019-present
# All rights reserved
# Licensed under a 3-clause BSD style license (see LICENSE)
__version__ = '3.4.1'
```

**Review by Kaderinho — APPROVED** [review]
LGTM !

If I understand correctly, it means that `agent integrations install` will fail if python3 is not present on the host environment ? If it's the case, it might be worth to add a release note as it is kind of a breaking change imho

**Review by FlorentClarret — APPROVED** [review]

**Comment by alopezz** [comment]
> If I understand correctly, it means that `agent integrations install` will fail if python3 is not present on the host environment ? If it's the case, it might be worth to add a release note as it is kind of a breaking change imho

Makes sense, added!

**Comment by alopezz** [comment]
Can we get this merged? The failing test looks unrelated.

File: releasenotes/notes/downloader-force-py3-3fb56a62ce62285d.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+upgrade:
+  - |
+    Downloading and installing official checks with `agent integration install`
+    is no longer supported for agent installations that do not include an embedded
```
Comment (hestonhoffman) [inline releasenotes/notes/downloader-force-py3-3fb56a62ce62285d.yaml:12]:
```suggestion
    is no longer supported for Agent installations that do not include an embedded
```

**Review by hestonhoffman — APPROVED** [review]
Small nit

**Review by yzhan289 — APPROVED** [review]

**Pull request merged**